lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivationSpec.hs

```diff
@@ -282,6 +299,29 @@ prop_passphraseHashMalformed
 prop_passphraseHashMalformed pwd = monadicIO $ liftIO $ do
     checkPassphrase pwd (Hash mempty) `shouldBe` Left ErrWrongPassphrase
 
+prop_keyToHexTextJcliCompatible
+    :: WalletKey k
+    => Unencrypted (k 'RootK XPrv)
+    -> Property
+prop_keyToHexTextJcliCompatible (Unencrypted k) = monadicIO $ do
```

Comment:
This is calling `jcli` directly, so it is so-to-speak an integration test. We expect unit test to not require any 3rd party interaction so that they can be run in isolation if needed. Please, move the test :pray:

Reply:
Something like `Test.Integration.Jormungandr.Scenario.CLI.Key`… even though this test is not _completely_ related to the `key` command? 🤔

Reply:
We need to duplicate the `Arbitrary (Unencrypted key)`  instances to `cardano-wallet-jormungandr:integration`. But we still need to keep them here (`cardano-wallet-core:unit`) for the `unXPrvStripPub` roundtrips.

And we need to duplicate `getMnemonic` yet again because that weird type mismatch stops us from moving it to `test-utils`.

😶 🔫

Reply:
"that weird type mismatch" ?

Reply:
https://input-output-rnd.slack.com/archives/GBT05825V/p1581523877011800

```
/Users/anviking/IOHK/cardano-wallet/lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs:477:47: error:
    • Couldn't match type ‘cardano-wallet-core-2020.1.27:Cardano.Wallet.Primitive.Mnemonic.Mnemonic
                             12’
                     with ‘Cardano.Wallet.Primitive.Mnemonic.Mnemonic mw0’
      NB: ‘Cardano.Wallet.Primitive.Mnemonic.Mnemonic’
            is defined in ‘Cardano.Wallet.Primitive.Mnemonic’
                in package ‘cardano-wallet-core-2020.1.27’
          ‘cardano-wallet-core-2020.1.27:Cardano.Wallet.Primitive.Mnemonic.Mnemonic’
            is defined in ‘Cardano.Wallet.Primitive.Mnemonic’
                in package ‘cardano-wallet-core-2020.1.27’
      Expected type: Gen (Cardano.Wallet.Primitive.Mnemonic.Mnemonic mw0)
        Actual type: Gen
                       (cardano-wallet-core-2020.1.27:Cardano.Wallet.Primitive.Mnemonic.Mnemonic
                          12)
    • In the second argument of ‘(<$>)’, namely ‘genMnemonic @12’
      In the expression: Just . SomeMnemonic <$> genMnemonic @12
      In the expression: (3, Just . SomeMnemonic <$> genMnemonic @12)
    |
477 |                 , (3, Just . SomeMnemonic <$> genMnemonic @12)
    |                                               ^^^^^^^^^^^^^^^

```
when defining `genMnemonic` in the `test-utils` package.

Reply:
It's moved at least. Duplication of generators is a separate issue we can live with for the moment.